src/screens/IdentityNew.js

```diff
@@ -52,6 +59,18 @@ function IdentityNew({ accounts, navigation }) {
 		accounts.updateNewIdentity({ name });
 	};
 
+	const onSeedTextInput = inputSeedPhrase => {
+		setSeedPhrase(inputSeedPhrase);
+		const addressGeneration = () =>
+			brainWalletAddress(inputSeedPhrase)
+				.then(({ bip39 }) => {
+					setIsSeedValid(validateSeed(inputSeedPhrase, bip39));
+				})
+				.catch(() => setIsSeedValid(inputSeedPhrase, false));
+		const debouncedAddressGeneration = debounce(addressGeneration, 200);
```

Comment:
This debounce isn't working. I remember struggling a lot with it in the past (in `screens/AccountRecover.js`).
To verify what I said, add a `console.log` in  `addressGeneration`, you'll see that it's printed at every key stroke. I couldn't manage to make it work in a reasonable time in this component though :(.

Reply:
true, let me have a try.

Reply:
updated, in `debounce.js` we create new instance of `timeout` at each call of `debounce`.